### PR TITLE
chore: add calling network quality handler

### PR DESCRIPTION
**Pull request by alexandreferris**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Network Quality Changed handler was not implemented for Calling.

### Solutions

Creation of NetworkQualityChanged, its implementation and usage on AVS calls on CallManager.

### CODE REVIEW

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt [hunk 1]
```diff
@@ -208,13 +210,28 @@ actual class CallManagerImpl(
...
 
-        // TODO(calling): Network Quality handler
+        // Network Quality
+        scope.launch {
+            withCalling {
+                val onNetworkQualityChanged = OnNetworkQualityChanged()
+
+                wcall_set_network_quality_handler(
+                    inst = deferredHandle.await(),
+                    wcall_network_quality_h = onNetworkQualityChanged,
```
Comment (vitorhugods) [inline logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt:220]: We might need to call `keepingStrongReference()` here once #536 is merged.

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/scenario/OnNetworkQualityChanged.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        conversationId: String,
+        userId: String,
+        clientId: String,
+        quality: Int,
+        roundTripTimeInMilliseconds: Int,
+        upstreamPacketLossPercentage: Int,
+        downstreamPacketLossPercentage: Int,
+        arg: Pointer?
+    ) {
+        // Not yet implemented
```
[the remark this answers is not in the document]
Comment (vitorhugods) [reply logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/scenario/OnNetworkQualityChanged.kt:18]: Can we at least log it to see what's up?

**Review by vitorhugods — COMMENTED** [review]
We need to give this handler the same treatment as the others after #536 is merged.

Other than that, LGTM

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt (hunk 1 above)
Comment (alexandreferris) [reply logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt:220]: Yep! Just waiting for it to be merged.

**Pull request merged**